### ISSUE TITLE
Unified base component for physics related components

### ISSUE THREAD

**Issue by Gmod4phun** [Facepunch/sbox-issues#4425]
### What it is?

As far as I can tell, there is not a 'unified' (sorry for a lack of a better term) base component for gameobjects that should have physics. There is the `Rigidbody` component, as well as `ModelPhysics` component.

If I want to find all gameobjects that have physics, I will need to look for any object that has either a Rigidbody or ModelPhysics component, not sure if there are any other variants of physics components. I don't know if this is intentional, or if it would be better to have a unified base component for physics objects.

### What should it be?

If it makes sense, probably have a base component, maybe a `PhysicsComponent` that would serve as the base for all other components that add physics to gameobjects. With that, finding objects with physics would be more straightforward.

**Comment by garrynewman** [comment]
I totally agree. I think that RigidBody should be enough, and ModelPhysics should be creating RigidBody children.

I am not really happy with ModelPhysics. It feels icky and special, I don't like that, so I'm gonna be finding a way to make it less special.
